src/main/java/com/homeadvisor/kafdrop/KafDrop.java

```diff
@@ -107,19 +111,41 @@ public int getOrder()
          return Ordered.HIGHEST_PRECEDENCE + 10;
       }
 
-      @Override
-      public void onApplicationEvent(ApplicationEnvironmentPreparedEvent event)
-      {
-         final ConfigurableEnvironment environment = event.getEnvironment();
-         if (environment.containsProperty(SM_CONFIG_DIR))
-         {
-            Stream.of("kafdrop", "global")
-               .map(name -> readProperties(environment, name))
-               .filter(Objects::nonNull)
-               .forEach(iniPropSource -> environment.getPropertySources()
-                  .addBefore("applicationConfigurationProperties", iniPropSource));
-         }
-      }
+        @Override
+        public void onApplicationEvent(ApplicationEnvironmentPreparedEvent event)
+        {
+            final ConfigurableEnvironment environment = event.getEnvironment();
+
+            LOG.info("Initializing jaas config");
+            String env = environment.getProperty("kafka.env");
+            Boolean isSecured = environment.getProperty("kafka.isSecured", Boolean.class);
+            LOG.info("env: {} .Issecured kafka: {}", env, isSecured);
+            if (isSecured && Strings.isNullOrEmpty(env)) {
+                throw new RuntimeException("'env' cannot be null if connecting to secured kafka.");
```

Comment:
Seems like `IllegalArgumentException` or `IllegalStateException` would be a more appropriate exception here.